lightblue/src/main/java/org/esbtools/eventhandler/lightblue/LightblueDocumentEventRepository.java

```diff
@@ -60,20 +59,19 @@
      * or superseded), should be put back in the document event entity pool to be processed later.
      */
     private final int documentEventBatchSize;
-    private final Locking locking;
+    private final LockStrategy lockStrategy;
     private final Map<String, ? extends DocumentEventFactory> documentEventFactoriesByType;
```

Comment:
Suggestion: DocumentEventFactories class with 
  DocumentEventFactory get(String type) throws NoSuchElementException, so we don't have to check everywhere